### PR TITLE
Xcommon cmake full

### PR DESCRIPTION
**Pull request by danielpieczko**
Converted examples and tests to XCommon CMake.
App note documentation updated to reflect the new CMakeLists.txt and build commands.
Documentation still builds with xdoc.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,80 +1,173 @@
-@Library('xmos_jenkins_shared_library@v0.32.0') _
+// This file relates to internal XMOS infrastructure and should be ignored by external users
+
+@Library('xmos_jenkins_shared_library@develop') _
+
+def clone_test_deps() {
+  dir("${WORKSPACE}") {
+    sh "git clone git@github.com:danielpieczko/test_support"
```
Comment (ed-xmos) [inline Jenkinsfile:7]: should this be your fork?

File: Jenkinsfile [hunk 2]
```diff
@@ -1,80 +1,173 @@
...
+          steps {
+            println "Stage running on ${env.NODE_NAME}"
+
+            dir("${REPO}") {
+              checkout scm
+
+              dir("examples") {
+                withTools(params.TOOLS_VERSION) {
+                  sh 'cmake -G "Unix Makefiles" -B build'
+                  sh 'xmake -C build -j 8'
```
Comment (ed-xmos) [inline Jenkinsfile:54]: execs have 16 cores if you want to save second or two

File: examples/AN00246_xua_example/doc/rst/AN00246_xua_example.rst [hunk 3]
```diff
@@ -17,22 +17,28 @@ the xCORE.ai Multichannel (MC) Audio board (XK-AUDIO-316-MC).
 The Makefile
 ------------
 
-To start using ``lib_xua``, you need to add ``lib_xua`` to your Makefile::
+To start using the XMOS XUA library, you need to add ``lib_xua`` to the dependent module list
+in the CMakeLists.txt file. This application note uses the ``lib_board_support`` software
+repository to provide common code to configure the xCORE.ai Multichannel (MC) Audio board for
```
Comment (ed-xmos) [inline examples/AN00246_xua_example/doc/rst/AN00246_xua_example.rst:22]: xcore.ai I think although I can't keep track of branding changes

File: examples/AN00246_xua_example/src/app_xua_simple.xc [hunk 4]
```diff
@@ -39,12 +40,32 @@ clock clk_audio_mclk_usb            = on tile[0]: XS1_CLKBLK_1;   /* Master cloc
...
+        MCLK_48,
+        0,                      // pll_sync_freq (unused when driving fixed clock)
+        AUD_316_PCM_FORMAT_I2S,
+        32,                     // data bits
+        2                       // channels per frame
+};
 
-/* See hwsupport.xc */
-void ctrlPort();
+unsafe client interface i2c_master_if i_i2c_client;
```
Comment (ed-xmos) [inline examples/AN00246_xua_example/src/app_xua_simple.xc:54]: ah maybe not

File: tests/requirements.txt [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# python_version 3.11.6
+# python_version 3.12.1
```
Comment (ed-xmos) [inline tests/requirements.txt:1]: putting requirements in tests does make sense in most cases..

File: tests/xua_sim_tests/test_i2s_loopback/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,67 @@
+cmake_minimum_required(VERSION 3.21)
+include($ENV{XMOS_CMAKE_PATH}/xcommon.cmake)
+project(test_i2s_loopback)
+
+macro(get_json_list json_key out_list)
```
Comment (ed-xmos) [inline tests/xua_sim_tests/test_i2s_loopback/CMakeLists.txt:5]: This is nice..

File: tests/xua_sim_tests/test_mixer_routing_input.py [hunk 7]
```diff
@@ -22,13 +21,14 @@ def do_test(options, capfd, test_file, test_seed):
...
 
-    build_options = []
-    build_options += ["TEST_SEED=" + str(test_seed)]
+    seed_hdr = Path(__file__).parent / testname / "src" / "test_seed.h"
+    with open(seed_hdr, "w") as f:
+        f.write(f"#define TEST_SEED ({test_seed})")
 
     result = Pyxsim.run_on_simulator(
         binary,
+        cmake=True,
```
Comment (ed-xmos) [inline tests/xua_sim_tests/test_mixer_routing_input.py:30]: Ah is this why you had a fork? Nice feature if so - need to get that merged and let people know about it

File: tests/xua_sim_tests/test_mixer_routing_output/CMakeLists.txt [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+set(APP_HW_TARGET test_xs3_600.xn)
+
+set(XMOS_SANDBOX_DIR ${CMAKE_CURRENT_LIST_DIR}/../../../..)
+
+set(APP_INCLUDES src
+                 ${CMAKE_CURRENT_LIST_DIR}/../shared)
+
+set(APP_COMPILER_FLAGS -O3)
+
+include(${CMAKE_CURRENT_LIST_DIR}/../../../examples/deps.cmake)
```
Comment (ed-xmos) [inline tests/xua_sim_tests/test_mixer_routing_output/CMakeLists.txt:14]: Doesn't including it twice just overwrite the deps var?

**Review by ed-xmos — DISMISSED** [review]
Big piece of work. No show stoppers but a few optional comments

File: Jenkinsfile (hunk 1 above)
Comment (danielpieczko) [reply Jenkinsfile:7]: No, it is all merged to the xmos org now. I'll fix this.

**Review by ed-xmos — APPROVED** [review]

File: tests/xua_sim_tests/test_mixer_routing_output/CMakeLists.txt (hunk 8 above)
Comment (danielpieczko) [reply tests/xua_sim_tests/test_mixer_routing_output/CMakeLists.txt:14]: deps.cmake sets the variable and then test_deps.cmake appends to it.

**Review by shuchitak — DISMISSED** [review]
Looks great!!

File: examples/AN00248_xua_example_pdm_mics/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,25 @@
+cmake_minimum_required(VERSION 3.21)
+include($ENV{XMOS_CMAKE_PATH}/xcommon.cmake)
+project(app_xua_pdm_mics)
+
+set(APP_HW_TARGET mic_array_ref.xn)
```
Comment (xross) [inline examples/AN00248_xua_example_pdm_mics/CMakeLists.txt:5]: As part of the lib_mic_array work this should get updated.

File: tests/test_deps.cmake [hunk 10]
```diff
@@ -0,0 +1,3 @@
+list(APPEND APP_DEPENDENT_MODULES "lib_logging(3.2.0)"
+                                  "lib_random"
```
Comment (xross) [inline tests/test_deps.cmake:2]: hopefully this can be pinned soon.

**Review by xross — APPROVED** [review]
Looks good

**Comment by danielpieczko** [comment]
Needed to update hardware_test_tools, so I've also set the jenkins shared library and infr_apps versions as well.

**Review by xross — APPROVED** [review]

**Pull request merged**